improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+        """If coordinate has bounds, check points are equal to upper bound"""
+        if cube.coord(coord).bounds is not None:
+            upper_bounds = cube.coord(coord).bounds[..., 1]
+            if not (cube.coord(coord).points == upper_bounds).all():
+                self.errors.append(f"{coord} points should be equal to upper bounds")
+
+    def check_spot_data(self, cube, coords):
+        """Check spot coordinates"""
+        self.prod_type = "spot"
+        if "title" in cube.attributes:
+            if SPOT_TITLE_SUBSTR not in cube.attributes["title"]:
+                self.errors.append(
+                    f"Title attribute {cube.attributes['title']} is not "
+                    "consistent with spot data"
+                )
+
+        self._check_coords_present(coords, SPOT_COORDS)
+
+    def run(self, cube):
+        """Populates self-consistent interpreted parameters, or raises collated errors
+        describing (as far as posible) how the metadata are a) not self-consistent,
+        and / or b) not consistent with the Met Office IMPROVER standard.
+
+        Although every effort has been made to return as much information as possible,
+        collated errors may not be complete if the issue is fundamental. The developer
+        is advised to rerun this tool after each fix, until no further problems are
+        raised.
+        """
+
+        # 1) Interpret diagnostic and type-specific metadata, including cell methods
+        if cube.name() in ANCILLARIES:
+            self.field_type = self.ANCIL
+            self.diagnostic = cube.name()
+            if cube.cell_methods:
+                self.errors.append(f"Unexpected cell methods {cube.cell_methods}")
+
+        elif cube.name() in SPECIAL_CASES:
+            self.field_type = self.diagnostic = cube.name()
+            if cube.name() == "weather_code":
+                if cube.cell_methods:
+                    self.errors.append(f"Unexpected cell methods {cube.cell_methods}")
+            elif cube.name() == "wind_from_direction":
+                if cube.cell_methods:
+                    expected = CellMethod(method="mean", coords="realization")
+                    if len(cube.cell_methods) > 1 or cube.cell_methods[0] != expected:
+                        self.errors.append(
+                            f"Unexpected cell methods {cube.cell_methods}"
+                        )
+            else:
+                self.unhandled = True
+                return
+
+        else:
+            if "probability" in cube.name() and "threshold" in cube.name():
+                self.field_type = self.PROB
+                self.check_probability_cube_metadata(cube)
+            else:
+                self.diagnostic = cube.name()
+                try:
+                    perc_coord = find_percentile_coordinate(cube)
+                except CoordinateNotFoundError:
+                    coords = get_coord_names(cube)
+                    if any(
```

Comment:
Is this capturing the case where the name of the cube doesn't have "probability" or "threshold" in the name (and is wrong?) but there is still a threshold coordinate in the cube?

Reply:
I do find this section a bit hard to follow, maybe a few more comments?

Reply:
Also do we need to do any more checking to ensure the name of the cubes are correct?

Reply:
Again: I had to draw a line somewhere at "what is a reasonable level of 'wrong' to expect of an input file".  Is there a valid case (apart from reliability calibration tables, which I've explicitly excluded because they are intermediate processing tools not disseminated fields) where a threshold coordinate is present and there is not "probability" in the title?

If not, the developer is going to get a pretty clear output from `display_interpretation` that says "this is a realizations field" (I haven't checked, but I don't think the presence of an unexpected threshold coordinate would raise any interpreter errors) and realise there is something seriously wrong with their metadata at that stage.  Again, hopefully said developer will then go away and seek out the metadata standard to find out how to correctly name their probabilistic data field.

Reply:
Ok, I'm happy to leave this as it is